Integrations/Slack/Slack.py

```diff
@@ -394,17 +398,74 @@ def mirror_investigation():
 
 def long_running_loop():
     """
-    Runs in a long running container - checking for newly mirrored investigations.
+    Runs in a long running container - checking for newly mirrored investigations and answered questions.
     """
     while True:
+        error = ''
         try:
             check_for_mirrors()
+            check_for_answers()
+        except (requests.exceptions.ConnectionError, requests.exceptions.Timeout,
+                requests.exceptions.TooManyRedirects,
+                requests.exceptions.RequestException, requests.exceptions.SSLError) as e:
+            error = 'Could not connect to the Slack endpoint: {}'.format(str(e))
         except Exception as e:
             error = 'An error occurred: {}'.format(str(e))
             demisto.error(error)
-            demisto.updateModuleHealth(error)
         finally:
-            time.sleep(5)
+            if error:
+                demisto.updateModuleHealth(error)
+            time.sleep(1)
+
+
+def check_for_answers():
+    """
+    Checks for answered questions
+    """
+
+    integration_context = demisto.getIntegrationContext()
+    questions = integration_context.get('questions', [])
+    users = integration_context.get('users', [])
+    if questions:
+        questions = json.loads(questions)
+    if users:
+        users = json.loads(users)
+
+    for question in questions:
+        headers = {'Accept': 'application/json', 'Content-Type': 'application/json'}
+        body = {
+            'token': BOT_TOKEN,
+            'entitlement': question.get('entitlement')
+        }
+        res = requests.post(ENDPOINT_URL, data=json.dumps(body), headers=headers, verify=VERIFY_ENDPOINT)
+        if res.status_code != 200:
+            demisto.error('Slack - failed to poll for answers: {}, status code: {}'
+                          .format(res.content, res.status_code))
+            continue
+        answer = res.json()
+        if not answer:
+            continue
+        payload: dict = answer.get('payload')
+        if payload and isinstance(payload, list):
```

Comment:
in the line above you are defining payload to be `dict`, so maybe need to use `Union` here